logical/plugin/backend_server.go

```diff
@@ -0,0 +1,146 @@
+package plugin
+
+import (
+	"net/rpc"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/vault/logical"
+)
+
+type backendPluginServer struct {
+	broker  *plugin.MuxBroker
+	backend logical.Backend
+}
+
+func (b *backendPluginServer) HandleRequest(args *HandleRequestArgs, reply *HandleRequestReply) error {
+	conn, err := b.broker.Dial(args.StorageID)
+	if err != nil {
+		*reply = HandleRequestReply{
+			Error: plugin.NewBasicError(err),
+		}
+		return nil
+	}
+	c := rpc.NewClient(conn)
+	defer c.Close()
+
+	storage := &StorageClient{client: c}
+	args.Request.Storage = storage
+
+	resp, err := b.backend.HandleRequest(args.Request)
+	*reply = HandleRequestReply{
+		Response: resp,
+		Error:    plugin.NewBasicError(err),
+	}
+
+	return nil
+}
+
+func (b *backendPluginServer) SpecialPaths(_ interface{}, reply *SpecialPathsReply) error {
+	*reply = SpecialPathsReply{
+		Paths: b.backend.SpecialPaths(),
+	}
+	return nil
+}
+
+func (b *backendPluginServer) Logger(_ interface{}, reply *BackendLoggerReply) error {
```

Comment:
But the client doesn't make an RPC call for this function, so I don't think the server handler is needed.